peekingduck/pipeline/nodes/model/jde_mot/jde_files/multitracker.py

```diff
@@ -0,0 +1,546 @@
+# Modifications copyright 2021 AI Singapore
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#      https://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Original copyright (c) 2019 ZhongdaoWang
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+"""
+Script for creating Track and JDE model instance
+"""
+
+from typing import Any, Dict, List, Tuple
+from collections import deque
+import numpy as np
+import torch
+from peekingduck.pipeline.nodes.model.jde_mot.jde_files.models import (
+    Darknet,
+)
+from peekingduck.pipeline.nodes.model.jde_mot.jde_files import matching
+from peekingduck.pipeline.nodes.model.jde_mot.jde_files.basetrack import (
+    BaseTrack,
+    TrackState,
+)
+from peekingduck.pipeline.nodes.model.jde_mot.jde_files.utils.kalman_filter import (
+    KalmanFilter,
+)
+from peekingduck.pipeline.nodes.model.jde_mot.jde_files.utils.utils import (
+    non_max_suppression,
+    scale_coords,
+)
+
+# mypy: ignore-errors
+# pylint: disable=invalid-name, too-many-instance-attributes, too-many-locals, too-many-branches, too-many-statements, too-few-public-methods, arguments-differ, no-member, pointless-string-statement, useless-object-inheritance
```

Comment:
avoid if at all possible

Reply:
Moved linting error suppression magic comments to function/class level.